src/components/RepositoryReleasesChangelog.tsx

```diff
@@ -1,163 +1,72 @@
-		title === 'breaking changes' ? 'uppercase' : 'capitalize'
-
-	return (
-		<Box key={title}>
-			{shouldShowTitle && (
-				<Heading
-					as="h2"
-					size="xl"
-					bgColor="background3"
-					mb={4}
-					py={4}
-					textTransform={textTransform}
-					position="sticky"
-					top={0}
-				>
-					{title}
-				</Heading>
-			)}
-			<Box mb={4}>
-				{releaseGroup.map((processedReleaseChange: ProcessedRelease) => (
-					<ProcessedReleaseChangeDescription
-						key={`${title}-${processedReleaseChange.id}`}
-						repository={repository}
-						processedReleaseChange={processedReleaseChange}
-						mb={8}
-					/>
-				))}
-			</Box>
-		</Box>
-	)
-}
-
 const RepositoryReleasesChangelog = ({
 	repository,
 	fromVersion,
 	toVersion,
 }: RepositoryReleasesChangelogProps) => {
 	const [filteredReleases, setFilteredReleases] =
 		useState<Array<Release> | null>(null)
+	const [isFilteringReleases, setIsFilteringReleases] = useState<boolean>(false)
 
-	const { processedReleases, isProcessing } =
-		useProcessReleases(filteredReleases)
-
-	const {
-		data: releases,
-		isLoading,
-		isFetched,
-	} = useReleasesQuery({
+	const { data: releases, isFetching } = useReleasesQuery({
 		repository,
 		fromVersion,
 		toVersion,
 	})
 
 	useEffect(() => {
-		if (releases && fromVersion && toVersion) {
-			const newFilteredReleases = filterReleasesByVersionRange({
-				releases,
-				from: fromVersion,
-				to: toVersion,
-			}).sort((a, b) => compareReleasesByVersion(a, b, 'asc'))
+		setIsFilteringReleases(true)
+		void new Promise<Array<Release> | null>((resolve) => {
+			if (releases && fromVersion && toVersion) {
+				const newFilteredReleases = filterReleasesByVersionRange({
+					releases,
+					from: fromVersion,
+					to: toVersion,
+				}).sort((a, b) => compareReleasesByVersion(a, b, 'asc'))
+				resolve(newFilteredReleases)
+			} else {
+				resolve(null)
+			}
+		}).then((newFilteredReleases) => {
 			setFilteredReleases(newFilteredReleases)
-		} else {
-			setFilteredReleases(null)
-		}
-	}, [fromVersion, releases, toVersion])
-
-	const shouldShowProcessedReleaseTitle = (() => {
-		if (!processedReleases) {
-			return false
-		}
-
-		const groupsTitles = Object.keys(processedReleases)
 
-		return groupsTitles.length > 1 || !groupsTitles.includes('others')
-	})()
-
-	const sortedGroupTitles: Array<string> | null = processedReleases
-		? Object.keys(processedReleases).sort(compareReleaseGroupsByPriority)
-		: []
+			setIsFilteringReleases(false)
+		})
+	}, [fromVersion, releases, toVersion])
 
-	// TODO: simplify conditional renders with state machine
 	return (
 		<>
-			{isProcessing && (
-				<>
-					<Skeleton width="20%" height={8} mb={4} />
-					<TextSkeleton />
-				</>
+			{(isFetching || isFilteringReleases) && (
+				<Flex align="center" justify="center" height="100%">
+					<CircularProgress
+						isIndeterminate
+						size="8"
+						color="primary.400"
+						aria-label="Loading releases"
+					/>
+				</Flex>
```

Comment:
I think this is the part that controls the "spinner" in the changelog part of the page?

Can you add a comment like: "Changelog spinner" so that we can find it quickly later?

Reply:
Yes, this is the one. I can do that for sure!

I was actually thinking about showing a loading skeleton instead of the spinner. Let me change it so we can decide which option looks better.

Reply:
Change pushed, you can check the skeleton in the review environment.

Here you have both versions:

**With spinner**

https://user-images.githubusercontent.com/2677072/185387820-977d4b16-bb6a-4b5b-b5af-c183e5dec1e8.mp4

**With skeleton**

https://user-images.githubusercontent.com/2677072/185387851-6dfc7f43-e954-4f69-b55e-7939ebb04777.mp4

---

I prefer the skeleton since it helps guess what sort of content is coming, and it gives the feeling is loading faster.

Reply:
I've pushed another change. I'm actually using both the spinner and the skeleton now:

- if you select the repository through the UI, then the spinner will appear while fetching the releases, and disappear when it's ready for the user to pick the desired versions.
- if you land in the comparator with a preloaded repo + versions from the URL query, then the skeleton will appear while fetching and processing the releases, and disappear when it's ready to display the final changelog (no user interaction needed)

Let me know what you think!

Reply:
I like that we use the spinner _and_ the skeleton now. 😄 

I'm still seeing the flash of the red "processing releases" warning though, when I try it on the Vercel preview:

`/comparator?repo=renovatebot%2Frenovate&from=32.52.2&to=latest`

Reply:
Let me check that!

Reply:
@HonkingGoose It should be fine now!